DSharpPlus/Entities/Interaction/Components/DiscordSelectComponent.cs

```diff
@@ -33,42 +33,36 @@ namespace DSharpPlus.Entities
     /// </summary>
     public sealed class DiscordSelectComponent : DiscordComponent
     {
-        /// <summary>
-        /// The custom Id of this component.
-        /// </summary>
-        [JsonProperty("custom_id", NullValueHandling = NullValueHandling.Ignore)]
-        public new string CustomId { get; set; }
-
         /// <summary>
         /// The options to pick from on this component.
         /// </summary>
         [JsonProperty("options", NullValueHandling = NullValueHandling.Ignore)]
-        public DiscordSelectComponentOption[] Options { get; set; } = Array.Empty<DiscordSelectComponentOption>();
+        public IReadOnlyList<DiscordSelectComponentOption> Options { get; internal set; } = Array.Empty<DiscordSelectComponentOption>();
 
         /// <summary>
         /// The text to show when no option is selected.
         /// </summary>
         [JsonProperty("placeholder", NullValueHandling = NullValueHandling.Ignore)]
-        public string Placeholder { get; set; }
+        public string Placeholder { get; internal set; }
 
         /// <summary>
         /// The minimum amount of options that can be selected. Must be greater than zero and less than or equal to <see cref="MaximumSelectedValues"/>. Defaults to one.
         /// </summary>
         [JsonProperty("min_values", NullValueHandling = NullValueHandling.Ignore)]
-        public int? MinimumSelectedValues { get; set; }
+        public int? MinimumSelectedValues { get; internal set; }
 
         /// <summary>
         /// The maximum amount of options that can be selected. Must be greater than zero and or equal to <see cref="MinimumSelectedValues"/>. Defaults to 1.
         /// </summary>
         [JsonProperty("max_values", NullValueHandling = NullValueHandling.Ignore)]
-        public int? MaximumSelectedValues { get; set; }
+        public int? MaximumSelectedValues { get; internal set; }
```

Comment:
```suggestion
        public int? MaximumSelectedValues { get; internal set; }
        
        /// <summary>
        /// Whether this select can be used.
        /// </summary>
        [JsonProperty("disabled", NullValueHandling = NullValueHandling.Ignore)]
        public bool Disabled { get; internal set; }
```

Reply:
[Select menus can't be disabled](https://discord.dev/interactions/message-components#select-menu-object-select-menu-structure)

Reply:
They can

Reply:
https://github.com/discord/discord-api-docs/pull/3209